fuzz/fuzz_targets/uncompress2.rs

```diff
@@ -0,0 +1,118 @@
+//! This fuzzer is intended to find memory safety bugs and undefined behavior. The input entire
+//! input is processed, allowing for analysis of files of arbitrary size. It also speeds up
+//! coverage by disabling checksum validation and disregarding correctness of the results.
+//!
+//! This test must be run with `--features disable-checksum`. It's also suggested to initialize
+//! fuzzing with a corpus of real zlib/gzip files.
+//!
+//! libfuzzer uses LLVM sanitizers to detect some classes of bugs and UB. For detecting
+//! Rust-specific UB, use Miri. Once a corpus with suitable coverage has been built, you can run
+//! Miri against the corpus by executing:
+//! ```
+//! MIRIFLAGS=-Zmiri-disable-isolation cargo miri nextest run --bin uncompress2
+//! ```
+//! This assumes the corpus is located in the default directory of `corpus/uncompress2`. If it
+//! isn't, specify the corpus directory with the `ZLIB_RS_CORPUS_DIR` environment variable.
+#![cfg_attr(not(any(miri, test)), no_main)]
+
+use libfuzzer_sys::{fuzz_target, Corpus};
+use libz_rs_sys::{
+    gz_header, inflate, inflateEnd, inflateGetHeader, inflateInit2_, z_stream, zlibVersion,
+};
+use zlib_rs::{InflateFlush, ReturnCode};
+
+fuzz_target!(|input: &[u8]| -> Corpus { run(input) });
+
+fn run(input: &[u8]) -> Corpus {
+    if input.is_empty() {
+        return Corpus::Reject;
+    }
+
+    let mut stream = z_stream::default();
+
+    let err = unsafe {
+        inflateInit2_(
+            &mut stream,
+            15 + 32, // Support both zlib and gzip files.
+            zlibVersion(),
+            size_of::<z_stream>() as _,
+        )
+    };
+    assert_eq!(ReturnCode::from(err), ReturnCode::Ok);
+
+    let mut extra = vec![0; 64];
+    let mut name = vec![0; 64];
+    let mut comment = vec![0; 64];
```

Comment:
Maybe we can vary these buffer sizes, or whether they're present at all, but can save that for a later improvement.